protocol/contracts/lease/src/position/spec/mod.rs

```diff
@@ -91,3 +114,85 @@ where
         ContractError::broken_invariant_if::<Self>(!invariant, msg)
     }
 }
+
+#[cfg(test)]
+mod test_calc_borrow {
+    use currency::dex::test::StableC1;
+    use finance::{coin::Coin, duration::Duration, liability::Liability, percent::Percent};
+
+    use crate::error::ContractError;
+
+    use super::Spec;
+
+    type TestLpn = StableC1;
+
+    #[test]
+    fn downpayment_less_than_min() {
+        let spec = spec(1_000, 100);
+        let borrow = spec.calc_borrow_amount(99.into(), None);
+
+        assert!(matches!(
+            borrow,
+            Err(ContractError::InsufficientTrasactionAmount(_))
+        ));
+    }
+
+    #[test]
+    fn borrow_less_than_min() {
+        let spec = spec(1_000, 100);
+        let borrow = spec.calc_borrow_amount(100.into(), Some(Percent::from_percent(43)));
+
+        assert!(matches!(
+            borrow,
+            Err(ContractError::InsufficientTrasactionAmount(_))
+        ));
+    }
+
+    #[test]
+    fn lease_less_than_min() {
+        let spec = spec(1_000, 200);
+        let borrow_1 = spec.calc_borrow_amount(250.into(), None);
+        assert!(matches!(
+            borrow_1,
+            Err(ContractError::InsufficientAssetAmount(_))
+        ));
+
+        let borrow_2 = spec.calc_borrow_amount(550.into(), Some(Percent::from_percent(81)));
+        assert!(matches!(
+            borrow_2,
+            Err(ContractError::InsufficientAssetAmount(_))
+        ));
+    }
```

Comment:
please add a positive invocation using the same `spec` instance. We aim at catching the corner case values.